element_array_ephys/export/nwb.py

```diff
@@ -0,0 +1,545 @@
+    """
+
+    def __init__(self, lfp_electrodes_query, chunk_length: int = 10000):
+        """
+        Parameters
+        ----------
+        lfp_electrodes_query: element_array_ephys.ephys_no_curation.LFP
+        chunk_length: int, optional
+            Chunks are blocks of disk space where data are stored contiguously and compressed
+        """
+        self.lfp_electrodes_query = lfp_electrodes_query
+        self.electrodes = self.lfp_electrodes_query.fetch("electrode")
+
+        first_record = (
+                self.lfp_electrodes_query & dict(electrode=self.electrodes[0])
+        ).fetch(as_dict=True)[0]
+
+        self.n_channels = len(lfp_electrodes_query)
+        self.n_tt = len(first_record["lfp"])
+        self._dtype = first_record["lfp"].dtype
+
+        super().__init__(buffer_shape=(self.n_tt, 1), chunk_shape=(chunk_length, 1))
+
+    def _get_data(self, selection):
+
+        electrode = self.electrodes[selection[1]][0]
+        return (self.lfp_electrodes_query & dict(electrode=electrode)).fetch1("lfp")[
+            selection[0], np.newaxis
+        ]
+
+    def _get_dtype(self):
+        return self._dtype
+
+    def _get_maxshape(self):
+        return self.n_tt, self.n_channels
+
+
+def add_electrodes_to_nwb(session_key: dict, nwbfile: pynwb.NWBFile):
+    """
+    Add electrodes table to NWBFile. This is needed for any ElectricalSeries, including
+    raw source data and LFP.
+
+    Parameters
+    ----------
+    session_key: dict
+    nwbfile: pynwb.NWBFile
+    """
+
+    electrodes_query = probe.ProbeType.Electrode * probe.ElectrodeConfig.Electrode
+
+    session_electrodes = (
+            probe.ElectrodeConfig.Electrode & (ephys.EphysRecording & session_key)
+    ).fetch()
+
+    for additional_attribute in ["shank_col", "shank_row", "shank"]:
+        nwbfile.add_electrode_column(
+            name=electrodes_query.heading.attributes[additional_attribute].name,
+            description=electrodes_query.heading.attributes[
+                additional_attribute
+            ].comment,
+        )
+
+    nwbfile.add_electrode_column(
+        name="id_in_probe", description="electrode id within the probe",
+    )
+
+    for this_probe in (ephys.ProbeInsertion * probe.Probe & session_key).fetch(
+            as_dict=True
+    ):
+        insertion_record = (ephys.InsertionLocation & this_probe).fetch1()
+        if insertion_record:
+            [insertion_record.pop(k) for k in ephys.InsertionLocation.heading.primary_key]
+            insert_location = json.dumps(insertion_record, cls=DecimalEncoder)
+        else:
+            insert_location = "unknown"
+
+        device = nwbfile.create_device(
+            name=this_probe["probe"],
+            description=this_probe.get("probe_comment", None),
+            manufacturer=this_probe.get("probe_type", None),
+        )
+        shank_ids = np.unique((probe.ProbeType.Electrode & this_probe).fetch("shank"))
```

Comment:
Avoid using numpy functions when there is an equivalent python function
```suggestion
        shank_ids = set((probe.ProbeType.Electrode & this_probe).fetch("shank"))
```